kernel/kernel-api/src/main/java/io/delta/kernel/client/ExpressionHandler.java

```diff
@@ -45,4 +46,17 @@ ExpressionEvaluator getEvaluator(
         StructType inputSchema,
         Expression expression,
         DataType outputType);
+
+    /**
+     * Create a selection vector, a boolean type {@link ColumnVector}, on top of the range of values
+     * given in <i>values</i> array.
+     *
+     * @param values Array of initial boolean values for the selection vector. The ownership of
+     *               this array is with the caller and this method shouldn't depend on it after the
+     *               call is complete.
+     * @param from   start index of the range, inclusive.
+     * @param to     end index of the range, exclusive.
+     * @return A {@link ColumnVector} of {@code boolean} type values.
+     */
+    ColumnVector createSelectionVector(boolean[] values, int from, int to);
```

Comment:
Will this ever be called with a `from` value not equal to 0?

Did you consider simplifying the API to make this most-common-case (and, perhaps the only case) easier?